### DISCUSSION TITLE
Hardware transcoding not working in Plex LXC

### DISCUSSION THREAD

**Discussion by bdoell88** [tteck/Proxmox#2314] [Help Needed]
Hi, the short summary of the issue I'm experiencing is that hardware transcoding is not working properly by default.

What I mean by that is that I have used the Plex script and Plex does not use hardware transcoding unless I run `chmod 0777 /dev/dri/*` on the host, but that does not persist after reboots, so I'm thinking that isn't the "right" solution. 

Any suggestions on how to improve this? After running the command, hardware transcoding works great.

**Comment by bjoerrrrn** [comment]
Can you try to chown the devices to root:render please?

```
chown root:render /dev/dri/renderD128 -R
chown root:render /dev/dri/by-path -R
chown root:render /dev/dri/card0 -R
```

This should persist any reboots...

**Reply by bdoell88** [reply]
This does not seem to persist.

before reboot (after running the command above)
```
ls -l /dev/dri/
total 0
drwxrwxrwx 2 root render       80 Jan  9 21:41 by-path
crwxrwxrwx 1 root render 226,   0 Jan  9 21:41 card0
crwxrwxrwx 1 root render 226, 128 Jan  9 21:41 renderD128
```

after reboot:
```
ls -l /dev/dri/
total 0
drwxr-xr-x 2 root root         80 Jan 10 17:50 by-path
crw-rw---- 1 root video  226,   0 Jan 10 17:50 card0
crw-rw---- 1 root render 226, 128 Jan 10 17:50 renderD128
```

**Reply by bdoell88** [reply]
I just realized that the chown command doesn't seem to work.
i had already done the chmod command from my initial post and just did chown over top.

here is a screenshot you can see after a reboot i run ls -l and then do the three chown commands and the output from ls -l is still the same.

![image](https://github.com/tteck/Proxmox/assets/70822874/83e0222a-a8a8-46a1-9d26-dc4dce5ae130)

**Comment by waggs15** [comment]
I am also having this issue. I've tried all of the above commands but it seems the CPU is still doing the transcoding. I've got plex set to use an RTX 4000. I also have a quadro p400 in this server and neither are able to do the transcode. I'm suspecting it has something to do with the kernal and or GPU drivers that I've installed onto proxmox.

**Reply by waggs15** [reply]
I'll follow this up, I installed the Unmanic LXC as well and it's also having some issues, utilizing the quadro p400. I'm still on the belief that the kernal issue that I mentioned is the culprit but I don't know how to reverse that.

**Reply by bdoell88** [reply]
I'll see if I can do a fresh install of promox on the host I'm using this weekend. This is pretty much a fresh install already, but I'll redo it and see what happens and I'll list out all the commands I run. I'm probably missing something as I'm still fairly new to proxmox.

**Reply by bobloadmire** [reply]
did you guys ever figure this out? fresh install, I can select my gpu in plex settings, but transcoding is still software. unprivileged lxc

**Reply by tteck** [reply]
Are you a Plex Pass member?

**Reply by bobloadmire** [reply]
Yes...

**Comment by bobloadmire** [comment]
> Are you a Plex Pass member?

Also I was just reading currently there is a kernel issue?

**Comment by howetech** [comment]
There is a issue / bug with the latest proxmox 6.8 kernel:

https://forums.plex.tv/t/ubuntu-24-04-hw-transcoding/873765/48

Disabling HDR tone mapping solved it for me and I can't notice any difference in playback quality

**Reply by bobloadmire** [reply]
thanks for the tone mapping tip, that was super easy way to solve the issue
